handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -1062,7 +1062,15 @@ private SSLEngineResult wrap(ByteBufAllocator alloc, SSLEngine engine, ByteBuf i
     public void channelInactive(ChannelHandlerContext ctx) throws Exception {
         boolean handshakeFailed = handshakePromise.cause() != null;
 
-        ClosedChannelException exception = new ClosedChannelException();
+        final Exception exception;
+        if (handshakePromise.isDone()) {
+            exception = new ClosedChannelException();
+        } else {
+            // Closed before the handshake was done.
+            exception = new SSLHandshakeException("Connection closed during SSL handshake");
+            exception.initCause(new ClosedChannelException());
```

Comment:
It doesn't provide much value and produces too much logging output if we generate stack-trace twice from the same method. Consider making either `ClosedChannelException` or `SSLHandshakeException` stack-less in this case.